docs/proposals/job_controller.md

```diff
@@ -0,0 +1,191 @@
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+
+
+## Use Cases
+1. Ability to have a controller that will create a run-once pod.
+1. Ability to have a controller that will have an option to specify reaction on pod’s success/failure.
+1. Ability to specify number of restarts upon pod failure.
+1. Ability to specify maximum running time for a pod.
+1. Ability to create a repetitive job, eg. database backup, sending emails.
+1. Ability to create one time job to be run at a specified time, eg. system analysis.
+1. Ability to create a chain of jobs dependent one on another.
+
+
+## Motivation
+The main goal is to provide a new pod controller type, called Job, with the following
+characteristics:
+* Be able to start of one or several pods tracked as a single entity representing
+a one execution of this job. Final job execution status will reflect the completion
+status of all pods started for this run.
+* Be able to trigger a new job execution either on demand (triggered through server
+API by an external agent or by another subsystem), or periodically based on a time-based
+schedule defined in job specification part (using ISO8601 syntax notation).
+* Be able to restart a limited number of time failing containers of pods created
+by one job execution (using "Restart on failure" logic extended by a maximal restart count).
+* Be able to limit the execution time of pods created by a job.
+* Be able to provide status information about job executions, along with a (limited)
+history of previous job runs (containing start/stop times with completion exit code).
+* Be able to trigger additional jobs/actions based on a success/failure of a job.
+
+
+## Job
+Jobs, similarly to replication controller, are managed by a dedicated manager which
+takes care of all jobs existing in the system, performing the following tasks:
+* Executes a loop on the full list of defined jobs, calling on each a defined
+synchronization call-back, each execution loop being spaced by a configuration
+defined time interval (with default value of 5 seconds).
+* Performs a (recursive) watch of the registry directory key containing jobs
+definition. For each job newly defined or updated in the system, the same
+synchronization routine is invoked.
+
+The job manager provides a way for each job defined with schedule information to
+setup a dedicated timer for the next schedule time. This allows a better accuracy
+of start execution time for pods launched by jobs and remove dependency on the
+interval value set for the synchronization loop.
+
+In order to precisely track job pods, 2 specific labels are pushed in created pod
+metadata:
+* the job name ("jobName")
+* A tag ("jobTag", either a generated UID, or simply the schedule date of this
+job execution) allowing to group all pods started for a given job execution.
+
+Along with labels, specific environment variables are also dynamically added in
```

Comment:
I can imagine this being useful internally for pods running actual jobs, but I don't see why this might be useful for api consumers? What kind of information you'd like to see from there? I was thinking about naming pods in a way that it'll contain that kind of information, but it'll be more for information purposes, imho.